### PR TITLE
Applying feross/standard (main code, part 1)

### PR DESCRIPTION
**Pull request by shtukas**
**Section 1**

This commit applies to the code base the standard defined here: http://standardjs.com/

The package.json file has been updated to declare a new dependency, but the test script has not been updated yet.

The test script is currently

```
"test": "gulp compile && tap test"
```

and it will become in a later commit

```
"test": "standard && gulp compile && tap test"
```

We are not yet including `standard` as part of the test because the test would fail for the moment. See reasons below

**Section 2: Places where the code need to be updated but has not changed yet**

1. Anything under the folder `test` has not changed. This will be done in a later commit as I did not feel like changing the code and the tests in the same commit.

2. Places where the script highlighted that an exception was not handled have not been modified. This will come in a later commit.

3. "src/capiQueryBuilder.js:109:41: Unnecessary use of conditional expression for default assignment." Will be done in a later commit.

**Section 3: Summary**

This commit is the "trivial" part of the migration.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -1,78 +1,78 @@
...
+var eslint = require('gulp-eslint')
+var path = require('path')
+process.env.ARTEFACT_PATH = __dirname
+var riffraff = require('node-riffraff-artefact')
 
-var LAMBDA_SOURCE = 'src/*.js';
-var DEPLOY_SOURCE = 'conf/deploy.json';
-var CLOUDFORMATION_SOURCE = 'conf/cloudformation/*.yml';
+var LAMBDA_SOURCE = 'src/*.js'
+var DEPLOY_SOURCE = 'conf/deploy.yml'
```
Comment (LATaylor-guardian) [inline gulpfile.js]: I don't think you have resolved your commits correctly. This file has various regressions in it now.

**Comment by oilnam** [comment]
Are we sure we want to do this? 

It will pollute the entire app history (in term of git blame and so on), imposing a standard that we're not likely to follow in the future anyway (I don't care if we use semicolons at the end of a statement etc).

On top of this I thought the plan was to migrate to ScalaJS, so maybe we should focus on that instead. Just my 2 cents.

**Comment by shtukas** [comment]
> Are we sure we want to do this?

Request from @mchv

File: gulpfile.js (hunk 1 above)
Comment (shtukas) [inline gulpfile.js]: Ok, no worries. I will reapply the changes from the master branch version of `gulpfile.js` directly.

**Comment by oilnam** [comment]
> Request from @mchv

https://en.wikipedia.org/wiki/Argument_from_authority

;)

**Comment by shtukas** [comment]
> Argument from authority

:+1: (^_^)

**Comment by oilnam** [comment]
Yeah if you read the link it's a logical fallacy though, it means "because he said so" is not a good answer, even if Mariot turns out be almost always right :)

**Comment by shtukas** [comment]
I read it, and also new about it. I just meant that I like the fact that you pointed this out :)

File: gulpfile.js (hunk 1 above)
Comment (shtukas) [inline gulpfile.js]: Done. Should be all right now. Lesson learnt.

**Comment by LATaylor-guardian** [comment]
@oilnam The point is that whether you choose to follow the standard or not, this will enforce that everything remains common and standardised throughout the project. You're right, we will pollute the git history somewhat from this point onwards, but ideally this would have been done at the start so is rectifying a mistake.

We do intend to rewrite this in ScalaJS in the future, although we have no specific time frame or are in any rush to do so given other priorities and for no outward benefit from a product point of view.

And you're right, no one should do something 'because they are told to' so well done for questioning it! Especially on your day off - you must be bored ha 😆

**Comment by mchv** [comment]
Just to be clear, I thought trying to use `std` was more an exercise to see how much we need to change the current code than an alternative to the plan to use a typed language.

I think the main reason to not apply if we decide we are rewriting it soon with `ScalaJS`. Apart from the technical rewrite, the thing we need to be careful is the `latency`  impact

> it means "because he said so" is not a good answer
💯 we should discuss about it especially if we don't all agree

**Comment by LATaylor-guardian** [comment]
If all is good, the code looks good 👍

**Comment by shtukas** [comment]
Thanks @LATaylor-guardian !

**Pull request merged**